Rules.py

```diff
@@ -66,5 +66,62 @@ def global_rules(world):
     set_rule(world.get_location('Canyon Business Scrub'), lambda state: state.has('Ocean Title Deed'))
     set_rule(world.get_location('Song From Mask Salesman'), lambda state: state.has('Ocarina of Time'))
     set_rule(world.get_location('Remove the Cursed Mask'), lambda state: state.has('Ocarina of Time'))
-    set_rule(world.get_location('Tunnel Balloon From Observatory'), lambda state: state.has('Bow') or (state.form('Deku') and state.has('Magic Meter')))
-    set_rule(world.get_location('Tunnel Balloon From ECT'), lambda state: state.has('Bow') or (state.form('Deku') and state.has('Magic Meter')))
+    set_rule(world.get_location('Tunnel Balloon From Observatory'), lambda state: (state.has('Bow') and state.form('Human')) or (state.form('Deku') and state.has('Magic Meter')))
+    set_rule(world.get_location('Tunnel Balloon From ECT'), lambda state: (state.has('Bow') and state.form('Human')) or (state.form('Deku') and state.has('Magic Meter')))
+    # RevelationOrange started adding rules here (plus a few changes in rules above)
+    # location names used are mostly guesses and can absolutely be changed later
+    # item names used are also guesses, as are some state function names probly
+    # location names may be longer than necessary so as to be descriptive, like we can change 'South Clock Town Hookshot
+    # Ledge Rupee Chest' if we want lol
+    set_rule(world.get_location('Clock Tower Platform HP'), lambda state: state.form('Human') or state.form('Goron') or state.form('Zora') or state.can_reach(world.get_location('Clock Town Business Scrub')))
+    set_rule(world.get_location('Festival Tower Rupee Chest'), lambda state: state.has('Hookshot') or state.can_reach(world.get_location('Clock Town Business Scrub')))
+    set_rule(world.get_location('South Clock Town Hookshot Ledge Rupee Chest'), lambda state: state.has('Hookshot'))
+    set_rule(world.get_location('Bremen Mask From Guru Guru'), lambda state: state.form('Human'))
+    set_rule(world.get_location('Rosa Sisters HP'), lambda state: state.has('Kamaro Mask'))
+
+    # obviously, this and simliar 'tests' might not be necessary at all
+    set_rule(world.get_location('Adult Wallet from bank'), lambda state: True)
+    set_rule(world.get_location('Bank HP'), lambda state: True)
+
+    # I don't know that this item matters at all or if we should even have a spot for it
+    # but we can if we want, you get it the same way you get the code in the first cycle, you just have to be human to
+    # get the actual notebook from them I think
+    set_rule(world.get_location('Bomber Notebook'), lambda state: state.has('Magic Meter') and state.form('Deku') and state.form('Human'))
```

Comment:
Human Link is the only hard requirement, the other is `can_reach('pop NCT balloon')` or something.